google/cloud/ndb/model.py

```diff
@@ -711,7 +711,8 @@ def _properties_of(entity):
     """
     seen = set()
 
-    for level in (entity,) + tuple(type(entity).mro()):
+    entity_type = type(entities[0])  # assume all entities are same type
+    for level in entities + tuple(entity_type.mro()):
         if not hasattr(level, "_properties"):
```

Comment:
Hmm, looking at the next line, I want to see instead:

```python
        level_properties = getattr(level, "_properties", {})
        for prop in level_properties.values():
```

Reply:
Ok, any particular reason?

Reply:
eb7815d

Reply:
Probably a Zope-induced allergy to `hasattr` :).  But I also prefer to avoid `continue` where possible -- I've been bitten a coupl of times by sneaky bugs involving tricksy loop logic.